src/extension.ts

```diff
@@ -82,7 +83,22 @@ export async function activate(context: vscode.ExtensionContext) {
 
   const graph = makeGraph(graphConstructors);
   var thatMark: SelectionWithEditor[] = [];
-
+  var recordTestCase = false;
+  const cursorlessRecordTestCaseDisposable = vscode.commands.registerCommand(
+    "cursorless.recordTestCase",
+    () => {
+      console.log("Recording test case for next command");
+      recordTestCase = true;
+      // Empty clipboard to avoid accidentally cluttering fixtures
+      vscode.env.clipboard.writeText("");
+    }
+  );
+  const cursorlessSetNavigationMapDisposable = vscode.commands.registerCommand(
+    "cursorless.setNavigationMap",
+    (newNavigationMap: NavigationMap) => {
+      navigationMap = newNavigationMap;
+    }
+  );
```

Comment:
Not sure this one should be a command.  Maybe we could use a mock?  I'd be tempted for now to just check whether the given decoration is the same on the navigation map as it was during recording.  If it is, we don't need to change anything.  If it isn't, we can throw an error.  It's a bit fragile but I believe that color allocation is deterministic, and if it changes / isn't, we will see the error and can do something fancier

The advantage of not touching the token map is that it will make recording gifs easier I believe because then everything will look normal

Reply:
We're on the same page that this shouldn't be a command. Was more straightforward than mocking it, so I went with this so I could keep going.

I really like the idea of not having to touch the token map if it's not necessary, so let's go that route.

Reply:
What do you think is the best way to you get at the extensions internals for this, in this case `navigationMap`? I could think of a bunch of different ways but none seems much better than another.

Reply:
Ah right.  What if we added `navigationMap` to the graph (something like https://github.com/pokey/cursorless-vscode/commit/2963d73c7c71ca4e2a9cf9372a846c8835b271c0)?  Then we could use `makeGraph` to construct a graph in the test, and mock `makeGraph` to return our graph?  Something like that?

Reply:
Mocking `makeGraph` has been giving me some trouble, since the mock needs to be in place before the extension is activated -- which happens before any tests are run 🤔